09_elements.html

```diff
@@ -97,7 +97,227 @@
+      <p>All WebDriver implementations MUST support finding elements by XPath 1.0 [[!XPATH]] with the <a href="http://dev.w3.org/html5/spec/Overview.html#interactions-with-xpath-and-xslt">edits from section 3.3</a> of the [[!HTML51]] specification made. If no native support is present in the browser, a pure JS implementation MAY be used. When called, the returned values MUST be equivalent of calling "<a href="http://www.w3.org/TR/DOM-Level-3-XPath/xpath.html#XPathEvaluator-evaluate">evaluate</a>" function from [[DOM-LEVEL-3-XPATH]] with the result type set to "<a href="http://www.w3.org/TR/DOM-Level-3-XPath/xpath.html#XPathResult-ORDERED-NODE-SNAPSHOT-TYPE">ORDERED_NODE_SNAPSHOT_TYPE</a> (7).</p>
     </section>
   </section>
+
+  <section>
+   <h2 id=element-visibility>Element Visibility</h2>
+
+   <p>The visibility of a <a
+    href=http://w3c.github.io/dom/#element>Document element</a>
+    is guided by what is perceptually visible to the human eye.
+    An element's visibility does in this context thus not relate
+    in any way to the <code><a
+    href=http://dev.w3.org/csswg/css-box/#visibility>visibility</a></code>
+    or <code><a
+    href=http://dev.w3.org/csswg/css-box/#display0>display</a></code>
+    style properties [[!CSS3BOX]].
+
+   <p>The approach used by WebDriver to ascertain an element's
+    visibility is based on crude approximations about the element's
+    nature and relationship in the tree.  An element is in general
+    to be considered visible if any part of it is drawn on the
+    canvas within the bounderies of the viewport.
+
+   <p>An element's <dfn>element visibility</dfn>
+    is a boolean state where true signifies visibility and false
+    invisibility.  To compute the state on <var>element</var>:
+
+   <ol>
+    <li>If the attribute <code>hidden</code> is set, return false.
+
+    <li>If the computed value of the <code>display</code> style
+     property is "none", return false.
+
+     <!-- TODO(ato): Not really sure what this means, needs review: -->
+    <li>If it has a [[!CSS3-2D-TRANSFORMS]] or [[!CSS3-3D-TRANSFORMS]]
+     style property that gives a negative X or Y coordinates to
+     the canvas, return false.
+
+    <li>If <var>element</var> is the document's root element, that
+     is <code>document.documentElement</code>:
+
+     <ol>
+      <li>If the computed value of the <code>background-color</code>
+       property is "transparent", run these substeps:
+
+       <ol>
+        <li>If <var>element</var> is an HTML <code>HTML</code>
+         element [[!HTML51]], and the computed value of the
+         <code>background-color</code> style property of the first
+         <code>BODY</code> element descendant of the element in
+         <a
+         href=http://www.whatwg.org/specs/web-apps/current-work/multipage/infrastructure.html#tree-order>tree
+         order</a>, relative to that element, is also "transparent",
+         return false.
```

Comment:
The conclusion we came to in the WG meeting is wrong because of the [last comment](https://www.w3.org/Bugs/Public/show_bug.cgi?id=24634#c6) in that bug.  Basically `body`'s `background-color` is only looked at if `html`'s `background-image` is unset and `background-color` is "transparent".  See section 3.11.2 in the [CSS Backgrounds specification](http://dev.w3.org/csswg/css-backgrounds/#special-backgrounds) which explains this quite clearly.